rustls/src/lib.rs

```diff
@@ -490,15 +488,6 @@ pub mod version {
     pub use crate::versions::TLS13;
 }
 
-/// All defined key exchange groups appear in this module.
-///
-/// ALL_KX_GROUPS is provided as an array of all of these values.
-pub mod kx_group {
-    pub use crate::kx::SECP256R1;
-    pub use crate::kx::SECP384R1;
-    pub use crate::kx::X25519;
-}
```

Comment:
I'm not sure I fully understand why this module exists, so I've removed it for now. It might be something that should stay removed now that which KX groups are supported is a property of a specific crypto backend.

Reply:
The `ConfigBuilder` allows the caller to supply an explicit set of key exchange groups. This is part of a scheme to allow applications to select at compile-time which algorithms are used, and makes sure that the unused algorithms don't get compiled into the binary... but as part of that you have to actually supply the symbols that are linked to those algorithms somehow.